### PR TITLE
enable anonymous credentials

### PR DESCRIPTION
**Pull request by DyfanJones**
This PR enable anonymous credentials to be used when connecting to public s3 buckets: This PR address the following tickets:
- https://github.com/paws-r/paws/issues/489
- https://github.com/paws-r/paws/issues/512
- https://github.com/paws-r/paws/issues/478

### CODE REVIEW

**Comment by davidkretch** [comment]
This looks good. I'm surprised at how simple a change it actually turned out to ultimately be.

File: paws.common/R/signer_v4.R [hunk 1]
```diff
@@ -461,3 +464,12 @@ get_uri_path <- function(url) {
 
   return(uri)
 }
+
+# Clear down headers for anonymous credentials
+# https://github.com/aws/aws-sdk-go/blob/a7b02935e4fefa40f175f4d2143ec9c88a5f90f5/aws/signer/v4/v4_test.go#L321-L355
+anonymous_headers <- function(headers){
+  found = grepl("X-Amz-*", names(headers))
```
Comment (davidkretch) [inline paws.common/R/signer_v4.R:471]: I would fix these `=` to `<-` though.

File: paws.common/tests/testthat/test_signer_v4.R [hunk 2]
```diff
@@ -137,3 +137,38 @@ test_that("presign", {
   expect_equal(q[["X-Amz-Date"]], "19700101T000000Z")
   expect_equal(q[["X-Amz-Target"]], "prefix.Operation")
 })
+
+
+test_creds <- Credentials(
```
Comment (davidkretch) [inline paws.common/tests/testthat/test_signer_v4.R:142]: I feel like this should go inside the test, or call it `anonymous_test_creds`.

**Comment by DyfanJones** [comment]
I agree, I thought it would be a massive logic change. However, once I got my head around it. It proven to be quiet nice. I will make the changes and merge.

**Pull request merged**